backend/api/documents.py

```diff
@@ -0,0 +1,132 @@
+            return jsonify(
+                {"message": "You should be document creator to delete this document!"}
+            )
+
+    return jsonify({}), 200
+
+
+@document_api.route('/documents', methods=["GET"])
+@jwt_required()
+def get_documents() -> Tuple[Any, int]:
+    user_identifier = get_jwt_identity()
+    user: Dict = cast(Dict, mongo.find_user_by_id(ObjectId(user_identifier)))
+
+    documents = mongo.select_document(user["company"], user["_id"])
+
+    for document in documents:
+        document.pop("content")
+        document["_id"] = str(document["_id"])
+
+    return jsonify(documents), 200
+
+
+@document_api.route('/approve/<document_id>', methods=["POST"])
+@jwt_required()
+def approve_document(document_id: str) -> Tuple[Any, int]:
+    user_identifier = get_jwt_identity()
+    document: Dict = cast(Dict, mongo.find_document(document_id))
+    user: Dict = cast(Dict, mongo.find_user_by_id(ObjectId(user_identifier)))
+
+    if user["role"] not in [Role.LAWYER, Role.ECONOMIST]:
+        return jsonify({"message": "Invalid role for approving document!"}), 403
+
+    if user_identifier in document["approved"]:
+        return jsonify({"message": "Document already approved by you!"}), 403
+
+    document["approved"].append(user_identifier)
+    mongo.update_document(document_id, "approved", document["approved"])
+    mongo.update_document(document_id, "status", Status.AGREED)
+
+    return jsonify({}), 200
+
+
+@document_api.route('/sign/<document_id>', methods=["POST"])
+@jwt_required()
+def sign_document(document_id: str) -> Tuple[Any, int]:
+    user_identifier = get_jwt_identity()
+    document = cast(Dict, mongo.find_document(document_id))
+    user = cast(Dict, mongo.find_user_by_id(ObjectId(user_identifier)))
+
+    if (
+        user["role"] != Role.GENERAL_DIRECTOR
+        or document["status"] not in [Status.AGREED, Status.SIGNING]
+        or not mongo.is_approved_by_company(document_id, user["company"])
+    ):
+        return jsonify({"message": "Signing validation failed!"}), 403
+
+    if user_identifier in document["signed"] or len(document["signed"]) >= 3:
+        return jsonify({"message": "Already signed!"})
+
+    document["signed"].append(user_identifier)
+    mongo.update_document(document_id, "signed", document["signed"])
+    mongo.update_document(document_id, "status", Status.SIGNING)
+
+    return jsonify({}), 200
+
+
+@document_api.route('/archive/<document_id>', methods=["POST"])
+@jwt_required()
+def archive_document(document_id: str) -> Tuple[Any, int]:
+    user_identifier = get_jwt_identity()
+    document: Dict = cast(Dict, mongo.find_document(document_id))
+    user: Dict = cast(Dict, mongo.find_user_by_id(ObjectId(user_identifier)))
+
+    if (
+        document["company"] != user["company"]
+        or document["status"] != Status.SIGNING
+        or len(document["signed"]) < 2
+    ):
+        return jsonify({"message": "Sign document before archive!"}), 403
```

Comment:
Split into three block, re-think about http code and change messages to more clear